python/pyarrow/tests/test_schema.py

```diff
@@ -304,3 +306,37 @@ def test_schema_repr_with_dictionaries():
 two: int32""")
 
     assert repr(sch) == expected
+
+
+def test_type_schema_pickling():
+    cases = [
+        pa.int8(),
+        pa.string(),
+        pa.binary(),
+        pa.binary(10),
+        pa.list_(pa.string()),
+        pa.struct([
+            pa.field('a', 'int8'),
+            pa.field('b', 'string')
+        ]),
+        pa.time32('s'),
+        pa.time64('us'),
+        pa.date32(),
+        pa.date64(),
```

Comment:
timestamp types should be fine as well right?

Reply:
thought I had one, let me add